urlb5d5be4365_abmargb_oursim_ext_tgz-pJ8-br_edu_ufcg_lsd_oursim_experiments_RemoteNoFScenarioGeneratorJ8/src/br/edu/ufcg/lsd/oursim/experiments/LocalConcurrentScenarioGenerator.java

```diff
@@ -0,0 +1,227 @@
+import br.edu.ufcg.lsd.oursim.events.global.HaltCondition;
+import br.edu.ufcg.lsd.oursim.events.peer.PeerEvents;
+import br.edu.ufcg.lsd.oursim.events.worker.WorkerEvents;
+import br.edu.ufcg.lsd.oursim.fd.FailureDetectorOptInjector;
+import br.edu.ufcg.lsd.oursim.network.DefaultWANNetwork;
+import br.edu.ufcg.lsd.oursim.queue.ListEventProxy;
+import br.edu.ufcg.lsd.oursim.trace.DefaultTraceCollector;
+import br.edu.ufcg.lsd.oursim.util.Configuration;
+
+public class LocalConcurrentScenarioGenerator {
+
+	static int TASKS_DURATION = 10000; 
+	static double LOCALHOST_DELAY = 5.;
+	
+	static int MAX_WORKERS = 4;
+	static int[] TASKS = new int[]{1, 2, 4, 20, 100};
+	static long[] SECONDJOBDELAY = new long[]{0, 15 * TASKS_DURATION, 30 * TASKS_DURATION, 50 * TASKS_DURATION}; 
+	static int JOBS_PER_BROKER = 1;
+	
+	static String SCENARIO = "local-concurrent";
+	
+	static int[][] SEEDS = new int[][]{
+		new int[]{214338003, 944166635, 570782134, 526559949, 919594043, 214338003}, 
+		new int[]{413630306, 617652072, 984949403, 2098010699, 1519794858, 413630306}, 
+		new int[]{70395995, 2035636432, 1898631199, 1140526672, 787341018, 70395995}, 
+		new int[]{1146986511, 1886216422, 282759243, 1356582152, 538696788, 1146986511}, 
+		new int[]{1303202085, 1074684734, 2004358020, 30184504, 1102953255, 1303202085} 
+	};
+	
+	public static void main(String[] args) throws Exception {
+	
+		Properties properties = Configuration.createConfiguration(new Properties());
+		properties.put(Configuration.PROP_BROKER_MAX_REPLICAS, "3");
+		properties.put(Configuration.PROP_BROKER_MAX_SIMULTANEOUS_REPLICAS, "1");
+		properties.put(Configuration.PROP_USE_FAILURE_DETECTOR, Boolean.TRUE.toString());
+		properties.put(Configuration.PROP_BROKER_SCHEDULER_INTERVAL, "10000");
+		properties.put(Configuration.PROP_FAILURE_DETECTOR_NAME, "fixed");
+		
+		for (int taskIdx = 0; taskIdx < TASKS.length; taskIdx++) {
+			
+			int workersNo = MAX_WORKERS;
+			int tasksNo = TASKS[taskIdx];
+			long secondJobDelay = 0;
+			
+			run(properties, workersNo, tasksNo, secondJobDelay);
+		}
+		
+		for (int workersNo = 1; workersNo <= MAX_WORKERS; workersNo++) {
+			
+			int tasksNo = TASKS[TASKS.length - 1];
+			long secondJobDelay = 0;
+			
+			run(properties, workersNo, tasksNo, secondJobDelay);
+		}
+		
+		for (int delayIdx = 0; delayIdx < SECONDJOBDELAY.length; delayIdx++) {
+			
+			int workersNo = MAX_WORKERS;
+			int tasksNo = TASKS[TASKS.length - 1];
+			long secondJobDelay = SECONDJOBDELAY[delayIdx];
+			
+			run(properties, workersNo, tasksNo, secondJobDelay);
+		}
+		
+	}
+
+	private static void run(Properties properties, int workersNo, int tasksNo, long secondJobDelay)
+			throws JSONException, FileNotFoundException {
+		for (int i = 0; i < 5; i++) {
+			DefaultWANNetwork.setSeed(SEEDS[i]);
+			run(properties, workersNo, tasksNo, secondJobDelay, i);
+		}
+	}
+	
+	private static void run(Properties properties, int workersNo, int tasksNo, long secondJobDelay, int replica)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 47 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 98% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 155-178 into a separate method.